lib/std/zig/ast.zig

```diff
@@ -10,7 +10,8 @@ pub const TokenIndex = usize;
 pub const Tree = struct {
     source: []const u8,
     tokens: TokenList,
-    root_node: *Node.Root,
+    /// null on parse error (errors not empty)
+    root_node: ?*Node.Root,
```

Comment:
I don't think this should be optional. When there is an error, it's undefined. When there is no error, the pointer is valid.  Note that there would never be a reason to check whether it was null because that would be redundant information as whether or not there was an error parsing. Indeed, if you look at the diff of all the usage sites, it's always using `.?`.